js/classes/entities/foreground/weather_system.js

```diff
@@ -0,0 +1,182 @@
+     */
+    constructor(type, intensity, time) {
+        this.type = type;
+        this.intensity = intensity;
+        this.time = time;
+
+        // Rainy day or night
+        this.makeHeavens();
+
+        if (this.type === "rain") {
+            this.makeClouds();
+            this.makePrecipitation();
+        }
+        else if (this.type === "snow") {
+            this.makeClouds();
+            this.makePrecipitation();
+            this.addSurfaceSnow();
+        } else if (this.type === "cloudy") {
+            this.makeClouds();
+            // Default to sunny with white clouds
+        }
+        // Defaults to sunny/(night with moon) and no clouds.
+        else {
+            this.makeHeavens();
+        }
+    }
+
+    /**
+     * Adds a sun/moon and updates the sky color.
+     */
+    makeHeavens() {
+        if (this.time === "night") {
+            BG_COLOR = COLORS.SKY_DARK_GREY;
+            GAME.addEntity(new Sun(new Vector(Camera.SIZE.x - 2 * Sun.SCALED_SIZE, Sun.SCALED_SIZE - 100), Sun.MOON), -1);
+
+        } else {
+            switch (this.type) {
+                case "warm":
+                    BG_COLOR = COLORS.SKY_BLUE;
+                    GAME.addEntity(new Sun(new Vector(Camera.SIZE.x - 2 * Sun.SCALED_SIZE, Sun.SCALED_SIZE - 100), Sun.VILLAGE), -1);
+                    break;
+                case "hot":
+                    BG_COLOR = COLORS.SKY_HOT_SKY;
+                    GAME.addEntity(new Sun(new Vector(Camera.SIZE.x - 2 * Sun.SCALED_SIZE, Sun.SCALED_SIZE - 100), Sun.LAVA), -1);
+                    break;
+                case "cloudy":
+                    BG_COLOR = COLORS.SKY_GREY;
+                    break;
+                case "rain":
+                    BG_COLOR = COLORS.SKY_GREY;
+                    break;
+                case "snow":
+                    BG_COLOR = COLORS.SKY_SNOW_GREY;
+                    break;
+                default:
+                    break;
+            }
+        }
+
+    }
+
+    /**
+     * Adds clouds.
+     */
+    makeClouds() {
+        // Make sure to only have a number of clouds that is divisble by 100. For example, 15 is bad value but 25 is a good value.
+        const numOfClouds = [2, 5, 10, 20, 25, 50];
+        let aboveGroundLevel = 19;
+        // TODO add in dark clouds.
+        if (this.makeRain) {
+            // Add dark clouds
+        } else {
+            // Add white clouds
+        }
+
+        let cloudNum = numOfClouds[this.intensity];
+        // Make so many clouds of each type.
+        for (let cloudType = 0; cloudType < 3; cloudType++) {
+            // Choose a multiplier that spaces out the clouds across the whole zone nicely.
+            let spaceBetweenMultiplier = 0;
+            for (spaceBetweenMultiplier = 0; spaceBetweenMultiplier < 100; spaceBetweenMultiplier++) {
+                if (spaceBetweenMultiplier * cloudNum == 100) {
+                    break;
+                }
+            }
```

Comment:
All you're doing here is computing division from scratch, so you can simply set `spaceBetweenMultiplier = 100 / numOfClouds` instead of going through all this trouble. This would also allow you to select cloud numbers that don't divide 100.

Reply:
I see what you mean. I assume you mean cloudNum and not numOfClouds. Perhpas I will rename some variables here as well.

Reply:
Yes, I meant `cloudNum`. My mistake.